activation/activation.go

```diff
@@ -175,8 +266,23 @@ func (b *Builder) GetLastSequence(node types.NodeId) uint64 {
 	}
 	atx, err := b.db.GetAtx(*atxId)
 	if err != nil {
-		log.Error("wtf no atx in db %v", *atxId)
+		b.log.Error("wtf no atx in db %v", *atxId)
 		return 0
 	}
 	return atx.Sequence
 }
+
+func (b *Builder) GetPositioningAtx(epochId types.EpochId) (*types.ActivationTx, error) {
+	posAtxId, err := b.GetPositioningAtxId(epochId)
+	if err == nil {
+		posAtx, err := b.db.GetAtx(*posAtxId)
+		if err != nil {
+			return nil, fmt.Errorf("cannot find pos atx: %v", err.Error())
+		}
+		return posAtx, nil
+	} else if b.prevATX != nil {
+		return b.prevATX, nil
+	} else {
```

Comment:
error is returned

Reply:
I think @moshababo is saying that the case of `err != nil` should come before the other cases, however, this would complicate the flow.

If there was an error and `b.prevATX != nil` we want to return `b.prevATX`. We could do something like:
```golang
	if err != nil {
		if b.prevATX != nil {
			return b.prevATX, nil
		}
		return nil, fmt.Errorf("cannot find pos atx id: %v", err)
	} else {
		posAtx, err := b.db.GetAtx(*posAtxId)
		if err != nil {
			return nil, fmt.Errorf("cannot find pos atx: %v", err.Error())
		}
		return posAtx, nil
	}
```

Not sure it's much better. @moshababo @antonlerner wdyt?

Reply:
Yes, although the `else` clause can be omitted.

I'm saying it because that's the common practice in Go.